src/main/java/org/cascadebot/cascadebot/commands/music/PlayCommand.java

```diff
@@ -17,6 +18,18 @@
 
     @Override
     public void onCommand(Member sender, CommandContext context) {
+        VoiceChannel voiceChannel = sender.getVoiceState().getChannel();
+
+        if (context.getData().getMusic().getJoinOnPlay()) {
+            if (voiceChannel != null) {
+                if (context.getMusicPlayer().getConnectedChannel() != voiceChannel) {
+                    context.runOtherCommand("join", sender, context);
+                }
+            } else {
+                context.runOtherCommand("join", sender, context);
+            }
```

Comment:
Few things.
1. check if bot is already connected to something, if so don't run this code
2. We don't need to check if the user is in a vc at all. The play command already does that.